pkgdb/src/buildenv/buildenv-lockfile.cc

```diff
@@ -125,35 +127,46 @@ nixpkgsHttpsToGithubInput( std::string locked_url )
 /* -------------------------------------------------------------------------- */
 
 static void
-buildenvPackageFromCatalogDescriptor( const nlohmann::json &  jfrom,
-                                      std::string &&          installId,
-                                      std::string &&          system,
-                                      BuildenvLockedPackage & pkg )
+buildenvPackageFromV1Descriptor( const nlohmann::json &  jfrom,
+                                 std::string &&          installId,
+                                 std::string &&          system,
+                                 BuildenvLockedPackage & pkg )
 {
   pkg.installId = installId;
   pkg.system    = system;
 
-  std::string attrPath = jfrom["attr_path"];
-
-  // This would be more appropriately moved to `evalCacheCursorForInput` where
-  // we are introducing the concept of a flake, but that code won't know where
-  // the attrPath is coming from to make that detemination.
-  pkg.attrPath = splitAttrPath( "legacyPackages." + system + "." + attrPath );
-  pkg.priority = jfrom["priority"];
-  pkg.input    = resolver::LockedInputRaw();
-
-
-  // These attributes are needed by the current builder, and not included in the
-  // descriptor. This will not always be true, but also may not be required to
-  // build depending on the path taken for future environment builds.
-
-  std::string locked_url = jfrom["locked_url"];
-
-  // Convert first from https to github and then to flox-nixpkgs
-  // TODO: do this in one hop instead of two
-  pkg.input       = nixpkgsHttpsToGithubInput( locked_url );
-  pkg.input.attrs = flox::githubAttrsToFloxNixpkgsAttrs( pkg.input.attrs );
-  pkg.input.url   = nix::FlakeRef::fromAttrs( pkg.input.attrs ).to_string();
+  if ( jfrom.contains( "locked-flake-attr-path" ) )
+    {
+      LockedInstallable lockedInstallable = LockedInstallable();
+      jfrom.get_to( lockedInstallable );
+      pkg.attrPath  = splitAttrPath( lockedInstallable.lockedFlakeAttrPath );
+      pkg.priority  = resolver::DEFAULT_PRIORITY;
```

Comment:
should we scrape priority of packages as well?

Reply:
Wouldn't we have to add it as a descriptor field in the manifest? Yes probably at some point, not sure we need to add it now

Reply:
i mean i guess they should have a priority field in the descriptor and we scrape them.
we dont necessarily need to do that now i think but its missing, same level of urgency for me as defining custom systems per installable maybe a bit higher.